crates/api/src/sitemap.rs

```diff
@@ -0,0 +1,67 @@
+use actix_web::{
+  http::header::{self, CacheDirective},
+  web::Data,
+  HttpResponse,
+  ResponseError,
+};
+use chrono::{DateTime, FixedOffset};
+use lemmy_api_common::context::LemmyContext;
+use lemmy_db_schema::{source::post::Post, utils::DbPool};
+use lemmy_utils::{error::LemmyResult, settings::SETTINGS};
+use sitemap_rs::{url::Url, url_set::UrlSet};
+use std::io::Write;
+use tracing::{error, info};
+
+async fn generate_urlset<W: Write>(
+  pool: &mut DbPool<'_>,
+  writer: &mut W,
+  limit: i64,
+) -> LemmyResult<()> {
```

Comment:
Dont know why this writer is used, just return the UrlSet and then convert it to string.